vortex-datafusion/src/lib.rs

```diff
@@ -372,86 +372,83 @@ fn get_column_references(expr: &Expr) -> HashSet<String> {
 }
 
 /// Physical plan node for scans against an in-memory, possibly chunked Vortex Array.
-#[derive(Debug, Clone)]
+#[derive(Clone)]
 struct VortexScanExec {
-    array: Array,
+    array: ChunkedArray,
     scan_projection: Vec<usize>,
     plan_properties: PlanProperties,
 }
 
+impl Debug for VortexScanExec {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("VortexScanExec")
+            .field("array_length", &self.array.len())
+            .field("array_dtype", &self.array.dtype())
+            .field("scan_projection", &self.scan_projection)
+            .field("plan_properties", &self.plan_properties)
+            .finish_non_exhaustive()
+    }
```

Comment:
I went with custom impl because `Array` logs the entire buffer contents as part of its `Debug` impl

Stdlib has a gated unstable feature to log a field using the contents of a closure, so in the future you'd be able to do something like

```
        f.debug_struct("VortexScanExec")
            .field_with("array_length", |mut f| {
                 f.debug_struct("Array")
                     .field("length", &self.array.len())
                     .field("dtype", self.array.dtype())
                     .finish()
             })
```

https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html#method.field_with